src/ssh-compute.ts

```diff
@@ -0,0 +1,189 @@
+    let cmd;
+
+    if (command && script) {
+      throw new Error('Both `command` and `entrypoint` inputs set - Please select one.');
+    }
+
+    if (user) {
+      instanceName = `${user}@${instanceName}`;
+    }
+
+    cmd = [
+      'compute',
+      'ssh',
+      instanceName,
+      '--zone',
+      zone,
+      '--ssh-key-file',
+      `${ssh_keys_folder}/${SSH_PRIVATE_KEY_FILENAME}`,
+      '--quiet', // we need to ignore prompts from console
+      '--tunnel-through-iap',
+    ];
+
+    // Save public and private ssh keys to the temp folder
+    await fs.mkdir(ssh_keys_folder, { recursive: true });
+    await fs.writeFile(`${ssh_keys_folder}/${SSH_PUBLIC_KEY_FILENAME}`, ssh_public_key, {
+      mode: 0o600,
+    });
+    await fs.writeFile(`${ssh_keys_folder}/${SSH_PRIVATE_KEY_FILENAME}`, '', { mode: 0o600 });
+    for (const key of ssh_private_key.split(/(?=-----BEGIN)/)) {
+      await fs.appendFile(`${ssh_keys_folder}/${SSH_PRIVATE_KEY_FILENAME}`, key.trim() + '\n');
+    }
+
+    if (container) {
+      cmd.push('--container', container);
+    }
+
+    if (flags) {
+      const flagList = parseFlags(flags);
+      if (flagList) cmd = cmd.concat(flagList);
+    }
+
+    if (script) {
+      try {
+        await fs.access(script);
+      } catch (error: any) {
+        const message = 'Script can not be found. Check script input path.';
+        throw new Error(message);
+      }
+
+      const commandData = (await fs.readFile(script)).toString('utf8');
+      command = `bash -c \"${commandData}\"`; // eslint-disable-line no-useless-escape
+    }
+
+    if (sshArgs) {
+      cmd.push(`-- ${sshArgs}`);
+    }
+
+    // Install gcloud if not already installed.
+    if (!gcloudVersion || gcloudVersion == 'latest') {
+      gcloudVersion = await setupGcloud.getLatestGcloudSDKVersion();
+    }
+    if (!setupGcloud.isInstalled(gcloudVersion)) {
+      await setupGcloud.installGcloudSDK(gcloudVersion);
+    } else {
+      const toolPath = toolCache.find('gcloud', gcloudVersion);
+      core.addPath(path.join(toolPath, 'bin'));
+    }
+
+    // Authenticate gcloud SDK.
+    await setupGcloud.authenticateGcloudSDK();
+    const authenticated = await setupGcloud.isAuthenticated();
+    if (!authenticated) {
+      throw new Error('Error authenticating the Cloud SDK.');
+    }
+
+    // set PROJECT ID
+    if (projectId) {
+      await setupGcloud.setProject(projectId);
+    } else if (process.env.GCLOUD_PROJECT) {
+      await setupGcloud.setProject(process.env.GCLOUD_PROJECT);
+    }
+    // Fail if no Project Id is provided if not already set.
+    const projectIdSet = await setupGcloud.isProjectIdSet();
+    if (!projectIdSet) {
+      throw new Error('No project Id provided.');
+    }
+
+    // Install beta components if needed and prepend the beta command
+    if (installBeta) {
+      await setupGcloud.installComponent('beta');
+      cmd.unshift('beta');
+    }
+
+    const toolCommand = setupGcloud.getToolCommand();
```

Comment:
There's a shorter and more succinct pattern for this now: https://github.com/google-github-actions/deploy-cloudrun/blob/c5e1552096165af5a328a6b3a4a28a733ea9d072/src/deploy-cloudrun.ts#L223